crates/model/src/solver_competition.rs

```diff
@@ -8,6 +8,19 @@ use serde::{Deserialize, Serialize};
 use serde_with::serde_as;
 use std::collections::BTreeMap;
 
+/// As a temporary measure the driver informs the api about per competition data that should be
+/// stored in the database. This goes to the api through an unlisted and authenticated http endpoint
+/// because we do not want the driver to have a database connection.
+/// Once autopilot is handling the competition this will no longer be needed.
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct Request {
+    pub auction: AuctionId,
```

Comment:
The `AuctionId` is already part of the `SolverCompetition`. Should this field maybe be dropped to avoid accidental inconsistencies? Alternatively one could assert that the auction ids match.

Reply:
I noticed this too and agree it looks a little weird. The field should always be in SolverCompetition because that struct gets stored directly in the database. We can kind of treat it as an opaque blob in this api so I also added it to the request where it has more meaning. I don't want to drop either field but I can add an assert if they don't match.